src/integer/z/properties.rs

```diff
@@ -191,6 +192,176 @@ impl Z {
         unsafe { fmpz_mod(&mut out.value, &self.value, &modulus.value) };
         out
     }
+
+    /// Computes a pair `(root, exp)` s.t. `root^exp = self`
+    /// if `self` is a perfect power and can be represented via `root.pow(exp)`.
+    ///
+    /// Returns a pair `(root, exp)` if `root.pow(exp) = self` exists. Otherwise,
+    /// `None` is returned.
```

Comment:
What is the behaviour of 16 = 4^2 = (-4)^2= 2^4 = (-2)^4? which return can one expect, the documentation itself should make this clear.
Maybe clarify that only some pair is returned or something similar, up to your personal choice